src/com/hms/flexyosisoftconnector/OsisoftJsonPayload.java

```diff
@@ -381,15 +381,34 @@ public void completePayloadAttempt() {
                     + " Float, or String.");
           }
 
-          payload.append(PayloadBuilder.addContainerStartToOMFDataMessage(tagName, typeString));
-
-          payload.append(tagPayload.getPayload());
-
-          payload.append(PayloadBuilder.addContainerEndToOMFDataMessage());
+          switch (communicationType) {
+            case OSIsoftConfig.OMF:
+              payload.append(PayloadBuilder.addContainerStartToOMFDataMessage(tagName, typeString));
+              payload.append(tagPayload.getPayload());
+              payload.append(PayloadBuilder.addContainerEndToOMFDataMessage());
+              break;
+            case OSIsoftConfig.PI_WEB_API:
+              payload.append(tagPayload.getPayload());
+              break;
+            default:
+              Logger.LOG_SERIOUS(comsErrMsg);
```

Comment:
Not in the scope of this PR, but worth noting that `comsErrMsg` looks like it is a constant, therefore it should be a `static final` and be capitalized appropriately using ALL_CAPS.

Reply:
agreed, it should get changed to a static final in the next PR.